internal/filter/filterconfig/config.go

```diff
@@ -176,7 +182,7 @@ func (mp *MatchProperties) ValidateForSpans() error {
 
 // ValidateForLogs validates properties for logs.
 func (mp *MatchProperties) ValidateForLogs() error {
-	if len(mp.SpanNames) > 0 || len(mp.Services) > 0 || len(mp.SpanKinds) > 0 {
+	if len(mp.SpanNames) > 0 || len(mp.Services) > 0 || len(mp.SpanKinds) > 0 || len(mp.MetricNames) > 0 {
```

Comment:
According to the documentation, `metric_names` should be allowed only for metrics